app/views/comments/_comment_proper.html.erb

```diff
@@ -25,40 +25,55 @@
-            <%= image_tag("twitter-logo.svg", class: "icon-img", alt: "twitter logo") %></a>
-        <% end %>
-        <% if comment.user.github_username.present? %>
-          <a href="https://github.com/<%= comment.user.github_username %>" rel="noopener noreferrer" target="_blank">
-            <%= image_tag("github-logo.svg", class: "icon-img", alt: "github logo") %>
           </a>
+          <% if comment.user.twitter_username.present? %>
+            <a href="https://twitter.com/<%= comment.user.twitter_username %>" rel="noopener noreferrer" target="_blank">
+              <%= image_tag("twitter-logo.svg", class: "icon-img", alt: "twitter logo") %></a>
+          <% end %>
+          <% if comment.user.github_username.present? %>
+            <a href="https://github.com/<%= comment.user.github_username %>" rel="noopener noreferrer" target="_blank">
+              <%= image_tag("github-logo.svg", class: "icon-img", alt: "github logo") %>
+            </a>
+          <% end %>
+          <% if commentable_author_is_op?(commentable, comment) %>
+            <span class="op-marker"><%= get_ama_or_op_banner(commentable) %></span>
+          <% end %>
+          <%= render "comments/comment_date", decorated_comment: decorated_comment %>
         <% end %>
-        <% if commentable_author_is_op?(commentable, comment) %>
-          <span class="op-marker"><%= get_ama_or_op_banner(commentable) %></span>
-        <% end %>
-        <%= render "comments/comment_date", decorated_comment: decorated_comment %>
         <button class="dropbtn" aria-label="Toggle dropdown menu">
-          <%= image_tag("three-dots.svg", class: "dropdown-icon", alt: "Dropdown menu icon") %>
+          <%= image_tag("three-dots.svg", class: "dropdown-icon", alt: "Dropdown menu") %>
         </button>
         <div class="dropdown">
           <div class="dropdown-content">
-            <a href="<%= comment.path %>">
-              Permalink
-            </a>
+            <% unless should_be_hidden?(comment, @root_comment) %>
+              <a href="<%= comment.path %>">
+                Permalink
+              </a>
+            <% end %>
             <span class="comment-actions hidden" data-user-id="<%= comment.user_id %>" style="display: none;">
               <a href="<%= comment.path %>/settings" rel="nofollow" style="color:#0a0a0a;" data-no-instant>Settings</a>
             </span>
+            <% action = comment.hidden_by_commentable_user ? "Unhide" : "Hide" %>
+            <span class="comment-actions hidden" data-action="hide-button" data-commentable-user-id="<%= commentable.user_id %>" data-user-id="<%= comment.user_id %>" style="display: none; width: 100%;">
+              <button class="hide-comment" type="button" data-hide-type="<%= action.downcase %>" data-comment-id="<%= comment.id %>">
+                <%= action %>
+              </button>
```

Comment:
Another weird implementation thing I have about this: it's a button that sends a `fetch` request, that then runs `location.reload()` on success. I feel like there's a more "Rails" way of resolving this, like making this a `<form>` that sends a regular `PATCH` request. I tried for a few minutes to get the `<form>` to work but the CSS was messy and the form didn't work as well anyway.